qiskit/algorithms/quantum_time_evolution/variational/principles/real/implementations/real_mc_lachlan_variational_principle.py

```diff
@@ -39,25 +42,40 @@ def __init__(
     def _get_raw_metric_tensor(
         self,
         ansatz,
-        param_dict: Dict[Parameter, Union[float, complex]],
+        parameters: List[Parameter],
+        # param_dict: Dict[Parameter, Union[float, complex]],
     ):
         raw_metric_tensor_real = metric_tensor_calculator.calculate(
-            ansatz, list(param_dict.keys()), self._qfi_method
+            ansatz, parameters, self._qfi_method
         )
 
-        return raw_metric_tensor_real
+        return raw_metric_tensor_real * 0.25
 
     def _get_raw_evolution_grad(
         self,
         hamiltonian,
         ansatz,
-        param_dict: Dict[Parameter, Union[float, complex]],
+        parameters: List[Parameter],
     ):
 
-        raw_evolution_grad_imag = evolution_grad_calculator.calculate(
-            hamiltonian, ansatz, list(param_dict.keys()), self._grad_method, basis=-1j * Y
-        )
+        def raw_evolution_grad_imag(param_dict: Dict,
+                                    backend: Optional[Union[BaseBackend, QuantumInstance]] = None):
+            energy = ~StateFn(hamiltonian) @ StateFn(ansatz)
+            energy = PauliExpectation().convert(energy)
+            #TODO rewrite to be more efficient
+            if backend is not None:
+                energy = CircuitSampler(backend).convert(energy, param_dict).eval()
+            else:
+                energy = energy.assign_parameters(param_dict).eval()
```

Comment:
Sure. You can change the logic accordingly.
With these changes I now made sure that the operator is correct. This can of course be processed more efficiently.